src/ci/bin/test.inc.sh

```diff
@@ -789,20 +794,58 @@ cromwell::private::delete_sbt_boot() {
     rm -rf ~/.sbt/boot/
 }
 
+cromwell::private::install_adoptopenjdk() {
+    # https://adoptopenjdk.net/installation.html#linux-pkg-deb
+    sudo apt-get install -y wget apt-transport-https gnupg
+    wget -qO - https://adoptopenjdk.jfrog.io/adoptopenjdk/api/gpg/key/public |
+        sudo apt-key add -
+    echo "deb https://adoptopenjdk.jfrog.io/adoptopenjdk/deb $(
+            grep UBUNTU_CODENAME /etc/os-release | cut -d = -f 2
+        ) main" |
+        sudo tee /etc/apt/sources.list.d/adoptopenjdk.list
+    sudo apt-get update
+    sudo apt-get install -y adoptopenjdk-11-hotspot
+    sudo update-java-alternatives --set adoptopenjdk-11-hotspot-amd64
+}
+
+cromwell::private::install_sbt_launcher() {
+    # Install sbt launcher
+    # Non-deb package installation instructions adapted from
+    # - https://github.com/sbt/sbt/releases/tag/v1.4.9
+    # - https://github.com/broadinstitute/scala-baseimage/pull/4/files
+    curl --location --fail --silent --show-error "https://github.com/sbt/sbt/releases/download/v1.4.9/sbt-1.4.9.tgz" |
+        sudo tar zxf - -C /usr/share
+    sudo update-alternatives --install /usr/bin/sbt sbt /usr/share/sbt/bin/sbt 1
+}
+
+cromwell::private::install_docker_compose() {
+    # Install or upgrade docker-compose so that we get the correct exit codes
+    # https://docs.docker.com/compose/release-notes/#1230
+    # https://docs.docker.com/compose/install/
+    curl \
+        --location --fail --silent --show-error \
+        "https://github.com/docker/compose/releases/download/1.28.5/docker-compose-$(uname -s)-$(uname -m)" \
+        > docker-compose
+    sudo mv docker-compose /usr/local/bin
+    sudo chmod +x /usr/local/bin/docker-compose
+}
+
+cromwell::private::setup_pyenv_python_latest() {
+    # Make `python` whatever the most recent version of python installed
+    # Fixes cases where someone has set pyenv to override `python` to use an older `python2` instead of `python3`
+    pyenv global "$(pyenv versions --bare --skip-aliases | sort -t '.' -k1,1n -k2,2n -k3,3n | tail -n 1)"
+}
+
 cromwell::private::pip_install() {
     local pip_package
     pip_package="${1:?pip_install called without a package}"; shift
 
     if [[ "${CROMWELL_BUILD_IS_CI}" == "true" ]]; then
-        if [[ "${CROMWELL_BUILD_PROVIDER}" == "${CROMWELL_BUILD_PROVIDER_CIRCLE}" ]]; then
-            pip3 install "${pip_package}" "$@"
-        else
-            sudo -H "${PYTHON3_HOME}/bin/pip" install "${pip_package}" "$@"
-        fi
+        sudo -H "$(command -v pip3)" install "${pip_package}" "$@"
```

Comment:
Will this work in CircleCI? I'm not sure exactly why the conditional was there.

Reply:
I believe so. I'm not sure why it was there. Maybe expediency?

Btw, [like Travis CI](https://docs.travis-ci.com/user/languages/minimal-and-generic/#minimal) Circle CI has `pip3` installed on even their most [minimal ubuntu images](https://discuss.circleci.com/t/linux-machine-executor-images-october-q4-update/37847).

Bonus "suspenders": On Circle CI we run the function `cromwell::private::setup_pyenv_python_latest()` which makes sure that `pip` is run the Python 3 version, just in case some other nested dependency script tries to run the command without specifying a version.